src/env/node/git/git.ts

```diff
@@ -2116,8 +2129,19 @@ export class Git {
 		return this.git<string>({ cwd: repoPath }, 'symbolic-ref', '--short', ref);
 	}
 
-	tag(repoPath: string) {
-		return this.git<string>({ cwd: repoPath }, 'tag', '-l', `--format=${parseGitTagsDefaultFormat}`);
+	async tag(repoPath: string, ...args: string[]) {
+		try {
+			const output = await this.git<string>({ cwd: repoPath }, 'tag', ...args);
```

Comment:
In the removed lines I see a `-l` and a  `--format=${parseGitTagsDefaultFormat}` being passed as arguments, and we've replaced those with an `args[]` parameter that leaves the caller responsible for passing them in. A few questions/thoughts here:

1. Before it seems we were always using these two params, and now it looks like we're being more flexible. Are they not always needed?
2. I wonder if we can get away with having more readable params like an `options` objects param that has something like a `useDefaultFormat?: boolean` for example, among other things dependening on which flags we use, and then just construct the args in here based on what is flagged on in the options? I know we do use an open-ended `args` param in other git commands in this file, so maybe that pattern is better and more flexible - just wanted to pitch the idea and get your thoughts.

Reply:
> 1. Before it seems we were always using these two params, and now it looks like we're being more flexible. Are they not always needed?
> 2. I wonder if we can get away with having more readable params like an `options` objects param that has something like a `useDefaultFormat?: boolean` for example, among other things dependening on which flags we use, and then just construct the args in here based on what is flagged on in the options? I know we do use an open-ended `args` param in other git commands in this file, so maybe that pattern is better and more flexible - just wanted to pitch the idea and get your thoughts.

I'll try to answer both, let me know if you want me to get more in detail with anything. Before, the `tag` method was only used from `localGitProvider:getTags` method to obtain the list of tags. However, with these changes, I've split the responsibility to the different layers:

- `commands/git/tag`:  Validates the command steps and arguments (for most cases), such as an invalid name, non-empty strings, etc. This layer also call's the corresponding git provider service method with options (if necessary).
- `git/gitProviderService`: At this layer we only retrieve the git provider and call the method with the same args. This layer is only intended to validate the provider has the requested method.
- `env/node/git/localGitProvider` (or the equivalent for remotes): Is in charge of doing the pertinent git calls and operations given a set of parameters and options. If we're talking about remotes, it will perform API calls. In this case, it runs git with a certain set of arguments given some parameters. It is also in charge of catching and adding context to errors that may happen during the execution of those commands.
- `env/node/git/git`: Lastly, at this layer is where the most atomic action happens, that is calling git with a set of arguments and catching any error that may happen to transform it into manageable errors.

Yes, we could move the arguments and options to the lower git layer, but then it wouldn't be an atomic action. It's a little hard to distinguish when something should happen at `localGitProvider` and when in `git`. But I went with this distinction and I think it's okay so far. You shouldn't be calling `git` from anywhere else but git provider service, though. Right? So it makes sense to leave control of the git arguments to `localGitProvider`.

As for the `useDefaultFormat` I think it's definitely something we can put into `getTags`' options arg. :+1: 

I tried to explain myself, but if you still think we could organize it different, please reach out to me and we can go over this 😄

Reply:
Sounds good - thanks for the detailed explanation!

Reply:
I just reviewed what `parseGitTagsDefaultFormat` is used for, and without it `parseGitTags` method doesn't make much sense because it expects the tag data in a given format. I agree that providing a custom format to the `getTags` method makes it more flexible, but it would require the method to receive a custom tag format & parser.

This is doable, but since it goes out of the PR scope I would leave it for a separate task if you're okay with it

Reply:
That's fine with me, thanks for looking into it.